File: framework/ui/src/io/intino/alexandria/ui/displays/components/BaseDate.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.schemas.Highlight;
import io.intino.alexandria.ui.displays.notifiers.BaseDateNotifier;

public class BaseDate<DN extends BaseDateNotifier, B extends Box> extends AbstractBaseDate<DN, B> {

    public BaseDate(B box) {
        super(box);
    }

    public BaseDate<DN, B> highlight(String textColor, String backgroundColor) {
        notifier.refreshHighlight(new Highlight().textColor(textColor).backgroundColor(backgroundColor));
        return this;
    }

    public BaseDate<DN, B> highlight(String color) {
        return highlight(null, color);
    }

    public BaseDate<DN, B> clearHighlight() {
        return highlight(null, null);
    }

}

File: web-ui/src/displays/components/BaseDate.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractBaseDate from "../../../gen/displays/components/AbstractBaseDate";
import BaseDateNotifier from "../../../gen/displays/notifiers/BaseDateNotifier";
import BaseDateRequester from "../../../gen/displays/requesters/BaseDateRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class BaseDate extends AbstractBaseDate {

	constructor(props) {
		super(props);
		this.notifier = new BaseDateNotifier(this);
		this.requester = new BaseDateRequester(this);
	};

	refresh = (value) => {
	};
}

export default withStyles(styles, { withTheme: true })(BaseDate);
DisplayFactory.register("BaseDate", withStyles(styles, { withTheme: true })(BaseDate));